draft-ietf-tls-rfc8446bis.md

```diff
@@ -2245,7 +2245,8 @@ Finite Field Groups (DHE):
 {:br }
 
 Items in "named_group_list" are ordered according to the sender's
-preferences (most preferred choice first).
+preferences (most preferred choice first). The "named_group_list"
+MUST NOT contain any duplicate entries.
```

Comment:
When you specify a requirement on the sender, it always helps to specify how the receiver might react if the sender misbehaves.  In this case, I think it best to only *allow* the receiver to reject the handshake, rather than require it.  After all, I believe that some implementations are tolerant of that.

```suggestion
MUST NOT contain any duplicate entries.  A sender MAY abort a connection
with a fatal illegal_parameter alert if it detect a duplicate entry.
```

(I'm guessing the alert type here.)

More generally, do we want to do this sort of thing with all such lists, not just groups?  key exchange mode? certificate type? etc...

Reply:
yes, probably "MAY abort a connection with an `illegal_parameter` Alert"

Reply:
Belatedly, yes, I'd be completely fine with martin's suggestion.

Reply:
Martin I beleatedly noticed you change says that the sender may abort the connection, it's the recipient that may abort the connection. Please ensure you are ok with this.

Reply:
Yes, of course you are right :)